src/matrix_ops.hpp

```diff
@@ -34,4 +33,38 @@ requires(matrix<V, Cols>::is_square)
     (std::make_index_sequence<Cols>{});
 }
 
+template <Vector V, std::size_t Cols, Vector W, std::size_t ColsOther>
+requires(std::is_same<typename extend_to<V, 1>::type,
```

Comment:
Is this constraint necessary? I would think you only need to check that both matrices have the same scalar type.

Reply:
Mm, right, this is not really needed, since it is the elements that are copied. There will be a "preference" for one of the two types of Vectors but it's irrelevant